### ISSUE TITLE
Multiple CF installations

### ISSUE THREAD

**Issue by MoritzHorch** [alphagov/paas-billing#84]
Hello,

we have multiple CF installations running and want to store their billiable events in a single database. Is there official support for this?

Or are you aware of any issues that could occur when running three collectors (one for each cf installation) and one api server (to query the single database).

**Comment by tlwr** [comment]
We haven't, to my knowledge, tried it; theoretically, it should work fine; as in you will be able to store and query events from multiple CFs.

Collector is composed of multiple `eventcollector`s which gets `RawEvent`s from fetchers; so in a sense running a single collector instance is running multiple collectors of events in parallel already.

I would be interested in knowing how you get on 🚀

**Comment by MoritzHorch** [comment]
Thanks for the answer, I just thought about it again and came to the conclusion that it will probably run into issues, due the way the CF API works. To avoid fetching the same events, you probably query the API with the "after_guid" parameter? If so, the chance of skipping events or running into errors is rather high. For example: If one CF installation write his events into the database, then another CF installation will take the guid of the event for his next API query and CF installation will return an error, as it does not know this guid. If you think otherwise, please let me know :)

Also: Is there a way to use this tool to only persist the app usage events, because that is what we are most interested. So we do not want to configure the pricing plans and so on, just want to have the relation app_usage_events and events handy. Is this possible?

**Comment by tlwr** [comment]
You are correct about `after_guid`, although it might end up re-fetching events in bulk rather than failling, which is not great...

Right now the collectors which are started are hardcoded, so you would have to do code modifications to get the behaviour you describe.

**Comment by MoritzHorch** [comment]
Thanks for the help, for a workaround I commented some lines in the SQL files to avoid errors. Not the nicest solutions, but `events` and `app_usage_events` are being collected now without the need to configure a correct pricing plan. 

If anyone is wondering, I achieved this by commenting 
`INSERT INTO billable_event_components_temp (select * from generate_billable_event_components());` in `create_billable_event_components.sql`. This will avoid errors such as `{"error":"/home/vcap/app/eventstore/sql/create_billable_event_components.sql: null value in column \"duration\" violates not-null constraint: Failing row contains`, but the collector will still keep collecting the (usage) events.
